src/dataflow/internal/process/functions/function-call.ts

```diff
@@ -107,6 +110,32 @@ export function processFunctionCall<OtherInfo>(functionCall: RFunctionCall<Other
 		inIds.push(...functionName.in, ...functionName.unknownReferences)
 	}
 
+	// parse a source call and analyze the referenced code
+	if(named && functionCallName == 'source') {
+		const sourceFile = functionCall.arguments[0]
+		if(sourceFile?.value?.type == RType.String) {
+			const executor = new RShellExecutor()
+			const path = removeTokenMapQuotationMarks(sourceFile.lexeme)
+
+			// parse, normalize and dataflow the sourced file
+			const parsed = executeSingleSubStep('parse', {
+				request:                'file',
+				content:                path,
+				ensurePackageInstalled: true
+			}, executor) as string
+			const normalized = executeSingleSubStep('normalize', parsed, executor.getTokenMap(), undefined, fileNameDeterministicCountingIdGenerator(path)) as NormalizedAst<OtherInfo & ParentInformation>
+			const dataflow = processDataflowFor(normalized.ast, { ...data, environments: finalEnv })
+
+			// update our graph with the sourced file's information
+			// TODO just set finalEnv, use overwriteEnvironments or appendEnvironments? makes no difference in the current example
```

Comment:
the behavior is similar to that of an expression list. With the knowledge that the environment handles all set semantics, overwrite should be correct.